File: backend/internal/models/ticket_listing.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type TicketListing struct {
	// gorm.Model
	// ID            uuid.UUID       `gorm:"unique;type:uuid;primaryKey;default:uuid_generate_v4()" json:"id"`
	// Price         float64         `gorm:"not null" json:"price"`
	// Status        string          `gorm:"not null" json:"status"`
	// CreatedAt     time.Time       `json:"createdAt"`
	// UpdatedAt     time.Time       `json:"updatedAt"`
	// DeletedAt     *time.Time      `gorm:"index"`
	// TicketId      uuid.UUID       `gorm:"not null" json:"ticketId"`
	ID            uuid.UUID `gorm:"unique;type:uuid;primaryKey"`
	Price         float64   `gorm:"not null"`
	Status        string    `gorm:"not null"`
	CreatedAt     time.Time
	UpdatedAt     time.Time
	DeletedAt     *time.Time      `gorm:"index"`
	TicketId      uuid.UUID       `gorm:"not null;uniqueIndex"`
	Ticket        Ticket          `gorm:"foreignKey:TicketId" json:"-"`
	Conversations *[]Conversation `gorm:"foreignKey:TicketListingId"`
	Sale          *Sale           `gorm:"foreignKey:TicketListingId"`
}

File: frontend/lib/core/models/ticket_listing.dart
import 'package:weezemaster/core/models/ticket.dart';

class TicketListing {
  final String id;
  final double price;
  final String status;
  final String ticketId;
  final Ticket ticket;
  final DateTime createdAt;
  final DateTime updatedAt;

  TicketListing({
    required this.id,
    required this.price,
    required this.status,
    required this.ticketId,
    required this.ticket,
    required this.createdAt,
    required this.updatedAt,
  });

  factory TicketListing.fromJson(Map<String, dynamic> json) {
    return TicketListing(
      id: json['ID'],
      price: (json['Price'] as num).toDouble(),
      status: json['Status'],
      ticketId: json['TicketId'],
      ticket: Ticket.fromJson(json['Ticket']),
      createdAt: DateTime.parse(json['CreatedAt']),
      updatedAt: DateTime.parse(json['UpdatedAt']),
    );
  }
}